experimental/unicodeset_parser/src/parse.rs

```diff
@@ -1098,41 +1071,36 @@ where
         }
     }
 
-    // parses [0-9a-fA-F]{1..6}
-    fn parse_variable_length_hex(&mut self) -> Result<(String, usize)> {
-        let mut result = String::new();
-        let mut end_offset = 0;
-        while let Some(&(offset, c)) = self.iter.peek() {
-            if result.len() >= 6 || !c.is_ascii_hexdigit() {
-                break;
-            }
-            result.push(c);
-            end_offset = offset;
-            self.iter.next();
-        }
-        if result.is_empty() {
-            let (unexpected_offset, unexpected_char) = self.must_peek()?;
-            return Err(PEK::UnexpectedChar(unexpected_char).with_offset(unexpected_offset));
-        }
-        Ok((result, end_offset))
+    fn parse_hex_digits_into_char(&mut self, min: usize, max: usize) -> Result<(usize, char)> {
+        let first_offset = self.must_peek_index()?;
+        let end_offset = self.validate_hex_digits(min, max)?;
+
+        // safety: validate_hex_digits ensures that chars (including the last one) are ascii hex digits,
```

Comment:
nit: this isn't about "safety" but about not panicking

Reply:
thanks, removed all other mentions in this file as well